fundrebalancer/fundrebalancer.go

```diff
@@ -674,9 +675,50 @@ func (r *FundRebalancer) isGasAcceptable(ctx context.Context, txns []SkipGoTxnWi
 		return false, "", fmt.Errorf("parsing max gas cost threshold")
 	}
 
-	if gasCostUUSDC.Cmp(maxCost) > 0 {
+	if gasCostUUSDC.Cmp(maxCost) <= 0 {
+		return true, gasCostUUSDC.String(), nil
+	}
+
+	// No fund rebalancing timeout set
+	if *chainConfig.FundRebalancingTimeout == -1 {
+		return false, gasCostUUSDC.String(), nil
+	}
+
+	transfers, err := r.database.GetPendingRebalanceTransfersToChain(ctx, chainID)
+	if err != nil {
+		return false, "", fmt.Errorf("getting pending transfers: %w", err)
+	}
+
+	if len(transfers) == 0 || chainConfig.FundRebalancingTimeout == nil {
 		return false, gasCostUUSDC.String(), nil
 	}
 
-	return true, gasCostUUSDC.String(), nil
+	// Use oldest pending transfer
+	oldestTransfer := transfers[0]
+	for _, transfer := range transfers {
+		if transfer.CreatedAt.Before(oldestTransfer.CreatedAt) {
+			oldestTransfer = transfer
+		}
+	}
```

Comment:
i may just be misunderstanding here so lmk, but I don't entirely get why we are using the oldest pending transfers timestamp to determine if the rebalance we are trying to do has been tried before and it timed out due to gas being too high. 

the chain id in this function is the chain that they are rebalancing from (i.e funds are being sent from this chainID to another chain), unless I am misunderstanding, this chain id won't have rebalances _to_ it (chainID will always be an evm chain which we don't rebalance to) so the `r.database.GetPendingRebalanceTransfersToChain` would always be an empty list i think

also assuming im wrong/misunderstanding about the above and we do get back a list of transfers from this chain, then what if there just are no pending transfers? a pending transfer will only be in the db if a transfer has actually been initiated on chain (and therefore passed the gas check). it seems like if there are no pending transfers in the db (because gas is too high) the transfer the solver is trying to do will never time out because there are no transfers in the db to check the ts against.

maybe we should be inserting txs to the db in some `pending_gas_approval` status, so we can know when we started trying to send the transfer and therefore know when it has timed out due to gas not being low enough (and then apply the cost cap)?